libdnf/dnf-repo.c

```diff
@@ -737,6 +753,101 @@ dnf_repo_set_keyfile(DnfRepo *repo, GKeyFile *keyfile)
 }
 
 /**
+ * dnf_repo_set_metadata_expire:
+ * @repo: a #DnfRepo instance.
+ * @metadata_expire: the expected expiry time for metadata
+ *
+ * Sets the metadata_expire time, which is default to be 0 days
+ **/
+void
+dnf_repo_set_metadata_expire(DnfRepo *repo, guint metadata_expire)
+{
+    DnfRepoPrivate *priv = GET_PRIVATE(repo);
+    priv->metadata_expire = metadata_expire;
+}
+
+/**
+ *  dnf_repo_parse_time_from_str
+ *  @expression: a expression to be parsed
+ *  @error: error item
+ *
+ *  parser to parse string into an integer value of seconds, or a human
+ *  readable variation specifying days, hours, minutes or seconds
+ *  until something happens. Works like :class:`BytesOption`.  Note
+ *  that due to historical president -1 means "never", so this accepts
+ *  that and allows the word never, too.
+ *
+ *  Valid inputs: 100, 1.5m, 90s, 1.2d, 1d, 0xF, 0.1, -1, never.
+ *  Invalid inputs: -10, -0.1, 45.6Z, 1d6h, 1day, 1y.
+
+ *  Return value will always be an integer
+ **/
+
+static guint dnf_repo_parse_time_from_str(const gchar *expression, GError **error)
+{
+    gchar last_char = expression[ strlen(expression) - 1 ];
+    gint multiplier;
+    guint result;
+    gdouble parsed_time;
+    gchar *endptr = NULL;
+
+    if (strlen(expression) < 1) {
```

Comment:
Very minor but i'd do `if (!*expression)` or if (!strcmp(expression, ""))` to avoid computing the length just for this check.